src/node/filesystem/binary.node.test.ts

```diff
@@ -11,23 +11,34 @@ const fixturePath = joinPath('fixtures', 'binary.txt');
 
 class MockStream extends Readable {
   private max: number;
+  private readSize: number;
   private index: number;
 
-  constructor(max = 10, options?: ReadableOptions) {
+  constructor(max = 10, readSize = 1, options?: ReadableOptions) {
     super(options);
     this.max = max;
-    this.index = 1;
+    this.readSize = readSize;
```

Comment:
praise: I really like how the readSize here allows the tests to cover both "remaining in internal buffer, stream still going" and "remaining in internal buffer, stream done"